app/jobs/subscribe_to_list_job.rb

```diff
@@ -0,0 +1,19 @@
+class SubscribeToListJob < ApplicationJob
+  queue_as :default
+
+  def perform(list_pardot_id, contact_salesforce_id)
+    @client = Pardot::Client.client
+
+    prospect_id = salesforce_to_prospect(contact_salesforce_id)
```

Comment:
it can but if it is, you shouldn't be here, changed to
`return unless prospect_id = Pardot.salesforce_to_prospect(subscription.contact.salesforce_id)`